Expose C-library

paretje opened the pull request:
Expose ms2pip as a C-library, allowing it to be used in the C-code of other projects.

paretje committed:
import changes from ionbot and define C API

paretje committed:
add get_include

paretje committed:
remove superfluous get_mz_ms2pip_ch2

paretje committed:
build clib

paretje committed:
specify destination of libms2pip

paretje committed:
build C-code for editable install

paretje committed:
ignore C-library

paretje committed:
move develop class out of setup.py

paretje committed:
expose MAX_PEPLEN in python

paretje committed:
rename cython_modules

RalfG commented on setup.py:
```diff
@@ -55,27 +58,30 @@
-]
-#_ = [[os.remove(f) for f in glob(pat)] for pat in to_remove]
+extra_compile_args = [
+        "-fno-var-tracking-assignments",
```
Drop for macOS support! :P

RalfG replied:
But I guess the merge should fix this

paretje committed:
rename ms2pip_pyx

paretje committed:
move models

paretje committed:
tune compile args

paretje committed:
Revert "move develop class out of setup.py"

This reverts commit 1b402daf1b81cb895c45b543d8314fa585fe3b7f. It creates
issues when installing using e.g. poetry, and I think I prefer this
small class definition above a sys path hack.

paretje committed:
ignore compiled pyx

paretje committed:
update MANIFEST

paretje committed:
add -Og

The build failure is due to memory consumption. This seems to be the
only change between `feature/library-ga` and earlier succeeding builds.
If this is the issue, I'm going to try to compensate this. For the test
builds, `-Og` isn't an issue, and should be set using `CFLAGS` to
speedup builds, but it might be nice to enable some optimisations for
production builds.

paretje committed:
install ms2pip.ms2pip_c
